### PR TITLE
Refactor of method signature for creating new windows

### PR DESCRIPTION
**Pull request by BenLambertNcl**
Change from window.open => ssf.window for opening new windows.

Reason for this change was because creating windows in OpenFin (using `fin.desktop.Window`) involves calling window.open, giving us an endless loop if we override window.open.

To make this change, I needed to add a browser API which is done by including the JavaScript files (the same as OpenFin).

### CODE REVIEW

File: packages/api-browser-demo/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "ssf-desktop-api-browser-demo",
+  "version": "0.0.1",
+  "description": "",
+  "main": "",
+  "scripts": {
+    "serve": "live-server --port=5000 src/",
+    "start": "npm run serve",
```
Comment (jleft) [inline packages/api-browser-demo/package.json]: Could probably remove `serve` and just have `start` as `live-server --port=5000 src/`

Not sure if it's worth specifying the port - could remove?

File: packages/api-browser-demo/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "postinstall": "rimraf src && copyfiles -f node_modules/ssf-desktop-api-specification/src/** src && copyfiles -f node_modules/ssf-desktop-api-browser/** src"
+  },
+  "author": "",
+  "license": "Apache-2.0",
+  "dependencies": {
+    "openfin-cli": "^1.1.5",
```
Comment (jleft) [inline packages/api-browser-demo/package.json]: Looks to be unused - remove.

File: packages/api-specification/src/new-window-api.html [hunk 3]
```diff
@@ -38,7 +38,7 @@
 <h3>Method Signature</h3>
 
 <pre>
-window.open(url, name, features)
+ssf.Window(url, name, features)
```
Comment (jleft) [inline packages/api-specification/src/new-window-api.html]: I think we'll need to consider whether the API will follow the factory pattern (used here), or the constructor pattern. For example:

```js
// Factory
var newWindow = ssf.window();

// Constructor
var newWindow = new ssf.Window();
```

From what I can see on the Wiki, there seems to be a preference for the constructor pattern, but I'm happy to use the factory pattern, get some feedback, then change if required.

However, if we are going to stick with the factory pattern for now, I think the casing should be changed. PascalCase for classes and constructors and camelCase for everything else is a common convention, so `ssf.Window()` => `ssf.window()`

File: packages/api-browser/window.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+if (!window.ssf) {
+  window.ssf = {};
+}
+
+window.ssf.Window = (url, name, features) => window.open(url, name, features);
```
Comment (jleft) [inline packages/api-browser/window.js]: Could use rest params here:

```js
(...args) => window.open(...args)
```

File: packages/api-openfin/window.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+if (!window.ssf) {
+  window.ssf = {};
+}
+
+window.ssf.Window = (url, name, features) => window.open(url, name, features);
```
Comment (jleft) [inline packages/api-openfin/window.js]: Could use rest params here:

```js
(...args) => window.open(...args)
```

**Review by jleft — COMMENTED** [review]
:+1: only a few style comments

File: packages/api-browser-demo/package.json (hunk 2 above)
Comment (BenLambertNcl) [inline packages/api-browser-demo/package.json]: It's used in `launch`, unless I'm mistaken?

File: packages/api-specification/src/new-window-api.html (hunk 3 above)
Comment (BenLambertNcl) [reply packages/api-specification/src/new-window-api.html]: Yeah sounds like a good idea. Won't be a big change if the feedback is to use the constructor pattern :)

**Review by jleft — APPROVED** [review]
LGTM - thanks!

**Pull request merged**